integration-libs/cdc/user-account/login-form/reconsent/cdc-reconsent.service.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable, OnDestroy } from '@angular/core';
+import { GlobalMessageService, GlobalMessageType } from '@spartacus/core';
+import { Subscription } from 'rxjs';
+import { LaunchDialogService } from '@spartacus/storefront';
+import { CdcJsService, CdcUserConsentService } from '@spartacus/cdc/root';
+
+@Injectable({ providedIn: 'root' })
+export class CdcReconsentService implements OnDestroy {
+  constructor(
+    protected cdcUserConsentService: CdcUserConsentService,
+    protected cdcJsService: CdcJsService,
+    protected globalMessageService: GlobalMessageService,
+    protected launchDialogService: LaunchDialogService
+  ) {}
+  protected subscription: Subscription = new Subscription();
+
+  /**
+   * saves the consent given from reconsent pop-up
+   * @param consentId - array of consent IDs
+   * @param userParams - data from login session
+   * @returns error message if any
+   */
+  saveReconsent(consentId: string[], userParams: any): string {
+    const errorMessage: string = '';
+    this.subscription.add(
+      this.cdcJsService.didLoad().subscribe((cdcLoaded) => {
+        if (cdcLoaded) {
+          consentId.forEach((consentTemplateId) => {
```

Comment:
Send all consents as an array to the preference object.Instead of one by one which will cause many unnecessary network calls

Reply:
done